tests/test_text.py

```diff
@@ -367,6 +381,69 @@ def test_candidate_citations(self):
             6,
         )
 
+    @parameterized.named_parameters(
+        [
+            dict(testcase_name="base-name", model="models/text-bison-001"),
+            dict(testcase_name="tuned-name", model="tunedModels/bipedal-pangolin-001"),
+            dict(
+                testcase_name="model",
+                model=model_types.Model(
+                    name="models/text-bison-001",
+                    base_model_id="text-bison-001",
+                    version="001",
+                    display_name="🦬",
+                    description="🦬🦬🦬🦬🦬🦬🦬🦬🦬🦬🦬",
+                    input_token_limit=8000,
+                    output_token_limit=4000,
+                    supported_generation_methods=["GenerateText"],
+                ),
+            ),
+            dict(
+                testcase_name="tuned_model",
+                model=model_types.TunedModel(
+                    name="tunedModels/bipedal-pangolin-001",
+                    base_model="models/text-bison-001",
+                ),
+            ),
+            dict(
+                testcase_name="glm_model",
+                model=glm.Model(
+                    name="models/text-bison-001",
+                ),
+            ),
+            dict(
+                testcase_name="glm_tuned_model",
+                model=glm.TunedModel(
+                    name="tunedModels/bipedal-pangolin-001",
+                    base_model="models/text-bison-001",
+                ),
+            ),
+            dict(
+                testcase_name="glm_tuned_model_nested",
+                model=glm.TunedModel(
+                    name="tunedModels/bipedal-pangolin-002",
+                    tuned_model_source={
+                        "tuned_model": "tunedModels/bipedal-pangolin-002",
+                        "base_model": "models/text-bison-001",
+                    },
+                ),
+            ),
+        ]
+    )
+    def test_count_message_tokens(self, model):
+        self.responses["get_tuned_model"] = glm.TunedModel(
+            name="tunedModels/bipedal-pangolin-001", base_model="models/text-bison-001"
+        )
+        self.responses["count_text_tokens"] = glm.CountTextTokensResponse(token_count=7)
+
+        response = text_service.count_text_tokens(model, "Tell me a story about a magic backpack.")
+        self.assertEqual({"token_count": 7}, response)
+        if len(self.observed_requests) > 1:
```

Comment:
Can we make the guard condition here something that makes it clear why it's optional? e.g. `if make_model_name(model).startswith("tunedModel/"): ...` or an extra param like `should_look_up_model=bool`

Reply:
Done.